homeassistant/components/insteon_plm/__init__.py

```diff
@@ -223,6 +233,24 @@ def _register_services():
                                schema=X10_HOUSECODE_SCHEMA)
         _LOGGER.debug("Insteon_plm Services registered")
 
+    def _fire_button_on_off_event(address, group, val):
+        # Firing an event when a button is pressed.
+        device = plm.devices[address.hex]
+        state_name = device.states[group].name
+        button = ("" if state_name == BUTTON_PRESSED_STATE_NAME
+                  else state_name[-1].lower())
+        schema = {CONF_ADDRESS: address.hex}
+        if button != "":
+            schema[EVENT_CONF_BUTTON] = button
+        if val:
+            _LOGGER.debug('Firing event %s.%s with address %s and button %s',
+                          DOMAIN, EVENT_BUTTON_ON, address.hex, button)
+            hass.bus.fire('{}.{}'.format(DOMAIN, EVENT_BUTTON_ON), schema)
+        else:
+            _LOGGER.debug('Firing event %s.%s with address %s and button %s',
+                          DOMAIN, EVENT_BUTTON_OFF, address.hex, button)
+            hass.bus.fire('{}.{}'.format(DOMAIN, EVENT_BUTTON_OFF), schema)
```

Comment:
Instead of formatting it on the fly, just store domain already in `EVENT_BUTTON_OFF`?